Sources/MongoSwift/MongoDatabase.swift

```diff
@@ -281,15 +351,37 @@ public class MongoDatabase {
     public func createCollection<T: Codable>(_ name: String,
                                              withType: T.Type,
                                              options: CreateCollectionOptions? = nil) throws -> MongoCollection<T> {
-        let encoder = BSONEncoder()
-        let opts = try encoder.encode(options)
+        let opts = try self.encoder.encode(options)
         var error = bson_error_t()
 
         guard let collection = mongoc_database_create_collection(self._database, name, opts?.data, &error) else {
             throw parseMongocError(error)
         }
 
-        return MongoCollection(fromCollection: collection, withClient: self._client)
+        let encoder = BSONEncoder(copies: self.encoder)
+        let decoder = BSONDecoder(copies: self.decoder)
+
+        if let (e, d) = options?.dateCodingStrategy?.rawValue {
```

Comment:
Thats a good idea. I've added the `CodingStrategyOptions` protocol for that purpose. I realized it may be useful for  users to configure their own encoders / decoders with these options, so I also defined a base `BSONCoderOptions` struct which conforms to that protocol.  

So now e.g. is possible:
```swift
let options = BSONCoderOptions(dateCodingStrategy: .secondsSince1970, dataCodingStrategy: .base64)
let encoder = BSONEncoder(options: options)
let decoder = BSONDecoder(options: options)
```
or, in the old way:
```swift
var encoder = BSONEncoder()
var decoder = BSONDecoder()
encoder.dateEncodingStrategy = .secondsSince1970
decoder.dateDecodingStrategy = .secondsSince1970
encoder.dataEncodingStrategy = .base64
decoder.dateDecodingStrategy = .base64
```